src/main/java/Scheduler.java

```diff
@@ -22,43 +24,121 @@ public Scheduler(ArrayList<Course> courses, ArrayList<Schedule> schedules) {
         this.schedules = schedules;
     }
 
+    public void addFilters(List<Filter> f) {
+        this.filters.addAll(f);
+    }
+
     /**
-     * Takes a list of course codes and outputs a schedule that takes the first lecture session and
-     * first tutorial session in each lecture.
+     * Scheduler used to create courses non recursively by permutation.
```

Comment:
Do this in Javadoc format, like the comment for `createBasicSchedule`. It'll also help the "`newCourses` should be sorted by priority" stick out